bundle/edu.gemini.lchquery.servlet/src/main/scala/edu/gemini/lchquery/servlet/LchQueryParam.scala

```diff
@@ -94,7 +94,14 @@ object LchQueryParam {
         Option(inv).map(i => s"${i.getFirst} ${i.getLast}").filterNot(_.trim.isEmpty)
 
       val spProg = toSPProg
-      (investigatorName(spProg.getGsaPhase1Data.getPi) :: spProg.getGsaPhase1Data.getCois.asScala.map(investigatorName).toList).
+      val piName = for {
+        f <- Option(spProg.getPIFirstName)
+        l <- Option(spProg.getPILastName)
+        n = s"$f $l".trim
+        if !n.isEmpty
```

Comment:
The `.trim` should reduce it to an empty `String` if both `f` and `l` are empty or spaces.  If one or the other is `null` though, the entire thing is skipped.  I'm not sure if that's intended.